src/server/ua_file.c

```diff
@@ -0,0 +1,419 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * 
+ *     Copyright (c) 2021 Kalycito Infotech Private Limited (Author: Jayanth Velusamy)
+ *
+*/
+
+#include <open62541/types.h>
+#include <open62541/plugin/log_stdout.h>
+#include "ua_server_internal.h"
+
+#ifdef UA_ENABLE_FILETYPE_OBJECT_SUPPORT
+
+#define FILE_CLOSE_STRING                                "Close"
+#define FILE_GETPOSITION_STRING                          "GetPosition"
+#define FILE_OPEN_STRING                                 "Open"
+#define FILE_READ_STRING                                 "Read"
+#define FILE_SETPOSITION_STRING                          "SetPosition"
+#define FILE_WRITE_STRING                                "Write"
+#define FILE_OPENCOUNT_STRING                            "OpenCount"
+#define FILE_SIZE_STRING                                 "Size"
+#define FILE_WRITABLE_STRING                             "Writable"
+#define FILE_USERWRITABLE_STRING                         "UserWritable"
+
+#define STATIC_QN(name) {0, UA_STRING_STATIC(name)}
+static const UA_QualifiedName fieldCloseQN = STATIC_QN(FILE_CLOSE_STRING);
+static const UA_QualifiedName fieldGetPositionQN = STATIC_QN(FILE_GETPOSITION_STRING);
+static const UA_QualifiedName fieldOpenQN = STATIC_QN(FILE_OPEN_STRING);
+static const UA_QualifiedName fieldReadQN = STATIC_QN(FILE_READ_STRING);
+static const UA_QualifiedName fieldSetPositionQN = STATIC_QN(FILE_SETPOSITION_STRING);
+static const UA_QualifiedName fieldWriteQN = STATIC_QN(FILE_WRITE_STRING);
+static const UA_QualifiedName fieldOpenCountQN = STATIC_QN(FILE_OPENCOUNT_STRING);
+static const UA_QualifiedName fieldSizeQN = STATIC_QN(FILE_SIZE_STRING);
+static const UA_QualifiedName fieldWritableQN = STATIC_QN(FILE_WRITABLE_STRING);
+static const UA_QualifiedName fieldUserWritableQN = STATIC_QN(FILE_USERWRITABLE_STRING);
+
+/* Gets the NodeId of a Field (e.g. Open) */
+static UA_StatusCode
+getFieldNodeId(UA_Server *server, const UA_NodeId *conditionNodeId,
+                        const UA_QualifiedName* fieldName, UA_NodeId *outFieldNodeId) {
+    UA_BrowsePathResult bpr =
+        UA_Server_browseSimplifiedBrowsePath(server, *conditionNodeId, 1, fieldName);
+    if(bpr.statusCode != UA_STATUSCODE_GOOD)
+    {
+        UA_LOG_ERROR(UA_Log_Stdout, UA_LOGCATEGORY_SERVER, "Parent NodeId not found!");
+        return bpr.statusCode;
+    }
+    UA_StatusCode retval = UA_NodeId_copy(&bpr.targets[0].targetId.nodeId, outFieldNodeId);
+    UA_BrowsePathResult_clear(&bpr);
+    return retval;
+}
+
+static UA_StatusCode
+updateOpenCountProperty(UA_Server *server, UA_FileType* fileObject)
+{
```

Comment:
Use the usual conventions of the entire library with curly brackets on the same line.